app/code/core/Mage/Adminhtml/Block/Sales/Order/Invoice/Create/Form.php

```diff
@@ -102,10 +102,7 @@ public function hasInvoiceShipmentTypeMismatch()
     public function canShipPartiallyItem()
     {
         $value = $this->getOrder()->getCanShipPartiallyItem();
-        if (!is_null($value) && !$value) {
-            return false;
-        }
-        return true;
+        return !(!is_null($value) && !$value);
```

Comment:
DeMorgan's Law `!(A && B) == (!A || !B)`:

```suggestion
        return is_null($value) || $value;
```